### DISCUSSION TITLE
no known leader error forever

### DISCUSSION THREAD

**Discussion by koh-osug** [canonical/dqlite#629] [Q&A] [answered]
I started 3 nodes and stopped them later. Now I'm starting again only the first node but it displays forever:

> 2024-03-09T20:45:25+01:00 | WARN  | attempt 3630: server 127.0.0.1:10001: no known leader
2024-03-09T20:45:25+01:00 | WARN  | attempt 3630: server 127.0.0.1:10003: dial: dial tcp 127.0.0.1:10003: connect: connection refused

How to restore the functionality of this single node again? Is this an error which will now be displayed forever or is the node even dysfunctional?

How can this automatically healed? When the other nodes are also starting again, will this be also automatically handled?

**Comment by cole-miller** [comment, marked as answer]
The first node that you restarted---call it A---is still operating with a configuration that includes all three nodes. Since it can't contact the other two nodes it can't win an election and become leader. For A to resume operating on its own you have to force it to commit a new configuration that cuts out the other two nodes. Instructions on how to do that are [here](https://microk8s.io/docs/restore-quorum) (written from the MicroK8s perspective but should be generally applicable, assuming you're using go-dqlite). Note that this will lead to data loss if A didn't have the latest database state when you took the cluster offline!

> When the other nodes are also starting again, will this be also automatically handled?

Yes, when the other nodes are restarted, they will find each other using the three-node configuration and eventually one of them will be able to win an election.

**Comment by koh-osug** [comment]
Thank you for the quick response.
